packages/firestore/test/unit/specs/write_spec.test.ts

```diff
@@ -394,6 +394,56 @@ describeSpec('Writes:', [], () => {
     );
   });
 
+  specTest(
+    'Held writes are not re-sent after disable/enable network.',
+    [],
+    () => {
+      const query = Query.atPath(path('collection'));
+      const docALocal = doc(
+        'collection/a',
+        0,
+        { v: 1 },
+        { hasLocalMutations: true }
+      );
+      const docA = doc('collection/a', 1000, { v: 1 });
+
+      return (
+        spec()
+          .userListens(query)
+          .watchAcksFull(query, 500)
+          .expectEvents(query, {})
+          .userSets('collection/a', { v: 1 })
+          .expectEvents(query, {
+            hasPendingWrites: true,
+            added: [docALocal]
+          })
+          // ack write but without a watch event.
+          .writeAcks(1000)
+
+          .disableNetwork()
+          .expectEvents(query, {
+            hasPendingWrites: true,
+            fromCache: true
+          })
+
+          // handshake + write + close = 3 requests
```

Comment:
This comment is a bit too terse. Is this happening as a result of disabling the network or something else?

If so it's weird because I would have thought when we disable the network nothing happens.

This could be a little clearer if you expected the write stream request count was 2 above and then this would show that all we did after that was close.

Reply:
Good point.  Added the requestCount==2 expectation prior to close.  FWIW, disabling the network sends an additional no-op write (https://github.com/firebase/firebase-js-sdk/blob/69c9a971fa4e48bf965369744559e4c9a2fab12b/packages/firestore/src/remote/persistent_stream.ts#L662). I don't know the full context of that, but I assume it's intentional (so the backend knows we got all the acks and can GC state probably).